Windows 上で mypy が platform 分岐を unreachable と判定して失敗する

## 背景
README/CLAUDE の開発手順に従い、Windows 環境で `mypy src/gfo` を実行した。

## 問題
Windows 上では `sys.platform == "win32"` が静的に真と見なされ、後続の非 Windows 分岐が unreachable と判定される。

エラー:
```text
src\gfo\config.py:78: error: Statement is unreachable  [unreachable]
Found 1 error in 1 file (checked 65 source files)
```

該当箇所は `get_config_dir()` の platform-specific path 解決。

## 期待する挙動
Windows のローカル開発でも `mypy src/gfo` が green になるようにしたい。

対応案:
- platform 分岐を mypy が unreachable と見なさない形にする
- または mypy config 側でこの platform-specific unreachable を許容する
- CI とローカルの想定 OS 差分を docs に明記する

## 調査メモ
関連箇所:
- `src/gfo/config.py` の `get_config_dir()`
- `pyproject.toml` の `warn_unreachable = true`

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -143,6 +143,11 @@ jobs:
       - name: Type check (mypy)
         run: uv run mypy src/gfo
 
+      # Windows ローカル開発 (README/CLAUDE の手順) でも green を保つため、
+      # sys.platform 分岐の unreachable を win32 視点でも検証する (issue #62)。
+      - name: Type check (mypy, --platform win32)
+        run: uv run mypy --platform win32 src/gfo
+
       # runner は 4 vCPU 固定なので xdist は -n 4 を明示 (auto でも同等だが意図を固定)。
       # カバレッジは max leg のみ取得する (matrix.cov)。min leg は cov 無し = 高速。
       # 3.13 (max) は coverage の sysmon コアを使い計測オーバーヘッドを抑える (3.11 は非対応)。

--- a/src/gfo/config.py
+++ b/src/gfo/config.py
@@ -75,10 +75,11 @@ def get_config_dir() -> Path:
         if appdata:
             return Path(appdata) / "gfo"
         return Path.home() / "AppData" / "Roaming" / "gfo"
-    xdg_config_home = os.environ.get("XDG_CONFIG_HOME", "").strip()
-    if xdg_config_home:
-        return Path(xdg_config_home) / "gfo"
-    return Path.home() / ".config" / "gfo"
+    else:  # mypy (win32) の platform 分岐枝刈りに載せるため明示的な else が必要
+        xdg_config_home = os.environ.get("XDG_CONFIG_HOME", "").strip()
+        if xdg_config_home:
+            return Path(xdg_config_home) / "gfo"
+        return Path.home() / ".config" / "gfo"
 
 
 def get_config_path() -> Path:

